manifests/init.pp

```diff
@@ -103,6 +103,21 @@
     tag      => 'splunk_server',
   }
 
+  if $facts['virtual'] == 'docker' {
+    ini_setting { 'OPTIMISTIC_ABOUT_FILE_LOCKING':
```

Comment:
Do you know if this is always needed in any Docker container? Or is it just required for the acceptance tests to work in Travis?

Reply:
Both.  I don't know the exact details but if you google *splunk docker unusable filesystem* This is pretty much the solution everyone provides.  Even the chef-splunk repo did this but this was for their acceptance test.  I Also could not find exact details on why/how this actually works.  If you want to try this out locally under docker remove that line and watch it fail with the above error message.

Reply:
Yeah, I saw the comment before. Just wondering whether it's best to do this in the module or in the acceptance test; I think a case could possibly be made either way, but seems kind of magical, esp. since there's no parameter to control it.

It sounds like if you setup the volumes correctly, it may not be necessary to set this?
https://hub.docker.com/r/alexanderm/docker-splunk/
```
To work around the unusable filesystem issue you need to mount in a volume from the host for the /opt/splunk/var/lib
For example I started docker like this:
docker run -v /data/splunk-test:/opt/splunk/var/lib -p 8000:8000 -it alexanderm/docker-splunk bash
```

Reply:
@wyardley I saw that but that isn't "Applicable" i believe in our case.  When running docker acceptance tests we are not taking a folder on the host and mapping it to a folder in the docker container.  Make sense?

Reply:
Ah, yeah, I thought this change was in the manifest  itself, not in the test. I agree that this is a reasonable workaround in the acceptance test itself.